### PR TITLE
WIP: EDTR: Added Apollo Client.

### PR DESCRIPTION
**Pull request by nicolad**
## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/src/components/TopLevelProviders/Apollo.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { ApolloProvider } from '@apollo/react-hooks';
+
+const cache = new InMemoryCache();
+const client = new ApolloClient( {
+	cache,
+	link: new HttpLink( {
+		uri: 'http://localhost:4000/graphql', // temp
+		headers: {
+			authorization: window.localStorage.getItem( 'token' ),
+			'client-name': 'Space Explorer [web]',
```
Comment (tn3rb) [inline assets/src/components/TopLevelProviders/Apollo.js]: Space Explorer ???

Comment (nicolad) [reply assets/src/components/TopLevelProviders/Apollo.js]: I think that's just a placeholder.

Comment (nicolad) [reply assets/src/components/TopLevelProviders/Apollo.js]: @tn3rb updated

**Comment by nicolad** [comment]
> looks good, but just a heads up that I'm working out a DDD inspired folder structure and will likely move this into something like `/src/infrastructure/`

:boom: done

File: assets/src/data/eventespresso/graphql-client.js [hunk 2]
```diff
@@ -1,19 +0,0 @@
-/**
```
Comment (nicolad) [inline assets/src/data/eventespresso/graphql-client.js]: This file isn't used anywhere, that's why it got removed.

File: assets/src/infrastructure/TopLevelProviders/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Internal dependencies */
```
Comment (nicolad) [inline assets/src/infrastructure/TopLevelProviders/index.js]: Here we might add some other providers, for example, `ThemeContext` from `styled-components` or similar, if any.

File: assets/src/data/eventespresso/graphql-client.js [hunk 4]
```diff
@@ -1,19 +0,0 @@
...
-	uri: graphqlEndpoint || '/graphql',
-	request: ( operation ) => {
-		operation.setContext( {
-			headers: {
-				'X-WP-Nonce': data.eejs_api_nonce,
-			},
-		} );
-	},
-} );
-
```
Comment (manzoorwanijk) [inline assets/src/data/eventespresso/graphql-client.js]: You can get the GQL endpoint URL from that.

Comment (nicolad) [reply assets/src/data/eventespresso/graphql-client.js]: Yes, updated later with those details: https://github.com/eventespresso/event-espresso-core/pull/1833/files#diff-e33616ff3b70812455391ad7eda7f428R11

**Comment by nicolad** [comment]
> Why is this PR to ` EDTR/refactor-master`?

Thanks for pointing that, updated the base branch.

File: assets/src/infrastructure/TopLevelProviders/Apollo.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { ApolloClient } from 'apollo-client';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { HttpLink } from 'apollo-link-http';
+import { ApolloProvider } from '@apollo/react-hooks';
+
+const { graphqlEndpoint, eejsdata: { data } } = window;
```
Comment (manzoorwanijk) [inline assets/src/infrastructure/TopLevelProviders/Apollo.js]: This is lost somewhere... That data not being set in PHP I guess.

Comment (nicolad) [reply assets/src/infrastructure/TopLevelProviders/Apollo.js]: Thanks, this is why I made it a WIP, because I knew that some details might be missing.

**Pull request closed without merging**